### PR TITLE
Publish PyTorch docs to pytorch/cpp repo

### PR DESCRIPTION
**Pull request by sekyondaMeta**
Updating the documents push to go to https://github.com/pytorch/docs repo instead of https://github.com/pytorch/pytorch.github.io as part of updating the PyTorch docs set up.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/documentation.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ body:
   attributes:
     label: 📚 The doc issue
     description: >
-      A clear and concise description of what content in https://pytorch.org/docs/stable/index.html is an issue. If this has to do with the general https://pytorch.org website, please file an issue at https://github.com/pytorch/pytorch.github.io/issues/new/choose instead. If this has to do with https://pytorch.org/tutorials, please file an issue at https://github.com/pytorch/tutorials/issues/new.
+      A clear and concise description of what content in https://pytorch.org/docs/stable/index.html is an issue. If this has to do with the general https://pytorch.org website, please file an issue at https://github.com/pytorch/docs/issues/new/choose instead. If this has to do with https://pytorch.org/tutorials, please file an issue at https://github.com/pytorch/tutorials/issues/new.
```
Comment (malfet) [inline .github/ISSUE_TEMPLATE/documentation.yml:9]: You don't need to change this one, as general website issues should still be reported against pytorch.github.io

File: .github/workflows/_docs.yml [hunk 2]
```diff
@@ -194,7 +194,7 @@ jobs:
           retention-days: 14
           s3-bucket: doc-previews
           if-no-files-found: error
-          path: pytorch.github.io/docs/main/
```
Comment (malfet) [inline .github/workflows/_docs.yml:197]: No, we probably want to keep this one as it's just for s3 bucket name prefix, and later do proper redirects in DrCI  (somehwere in https://github.com/pytorch/test-infra/tree/main/torchci )

**Review by malfet — COMMENTED** [review]
You'll need to either create a `site` branch in docs repo, or change 
```shell
branch="${3:-${DOCS_BRANCH:-site}}"
```
in `.ci/pytorch/python_doc_push_script.sh` to point to `main`

@clee2000 can you please configure OIDC to give pytorch permissions to push to that repo?

File: RELEASE.md [hunk 3]
```diff
@@ -218,7 +218,7 @@ Validate the release jobs for pytorch and domain libraries should be green. Vali
   * [TorchVision](https://hud.pytorch.org/hud/pytorch/vision/release%2F1.12)
   * [TorchAudio](https://hud.pytorch.org/hud/pytorch/audio/release%2F1.12)
 
-Validate that the documentation build has completed and generated entry corresponding to the release in  [docs folder](https://github.com/pytorch/pytorch.github.io/tree/site/docs/) of pytorch.github.io repository
+Validate that the documentation build has completed and generated entry corresponding to the release in  [docs repository](https://github.com/pytorch/docs/tree/site/docs/).
```
Comment (svekars) [inline RELEASE.md:221]:
```suggestion
Validate that the documentation build has completed and generated entry corresponding to the release in  [docs repository](https://github.com/pytorch/docs/tree/main/).
```

File: .ci/pytorch/python_doc_push_script.sh [hunk 4]
```diff
@@ -68,8 +68,8 @@ build_docs () {
 }
 
 
-git clone https://github.com/pytorch/pytorch.github.io -b "$branch" --depth 1
-pushd pytorch.github.io
+git clone https://github.com/pytorch/docs -b "$branch" --depth 1
+pushd docs
```
Comment (malfet) [inline .ci/pytorch/python_doc_push_script.sh:72]:
```suggestion
git clone https://github.com/pytorch/docs pytorch_docs -b "$branch" --depth 1
pushd pytorch_docs
```

File: .ci/pytorch/python_doc_push_script.sh [hunk 5]
```diff
@@ -68,8 +68,8 @@ build_docs () {
 }
 
 
-git clone https://github.com/pytorch/pytorch.github.io -b "$branch" --depth 1
-pushd pytorch.github.io
+git clone https://github.com/pytorch/docs pytorch_docs -b "$branch" --depth 1
+pushd pytorch_docs
```
Comment (malfet) [inline .ci/pytorch/python_doc_push_script.sh:72]:
```suggestion
mkdir -p pytorch_docs && pushd pytorch_docs
git clone https://github.com/pytorch/docs  -b "$branch" --depth 1
```

File: .ci/pytorch/python_doc_push_script.sh [hunk 6]
```diff
@@ -26,7 +26,7 @@ echo "error: python_doc_push_script.sh: version (arg2) not specified"
 fi
 
 # Argument 1: Where to copy the built documentation to
-# (pytorch.github.io/$install_path)
+# (docs/$install_path)
 install_path="${1:-${DOCS_INSTALL_PATH:-docs/${DOCS_VERSION}}}"
```
Comment (malfet) [inline .ci/pytorch/python_doc_push_script.sh:30]:
```suggestion
install_path="${1:-${DOCS_INSTALL_PATH:-${DOCS_VERSION}}}"
```

File: .github/ISSUE_TEMPLATE/documentation.yml (hunk 1 above)
Comment (malfet) [inline .github/ISSUE_TEMPLATE/documentation.yml:9]:
```suggestion
      A clear and concise description of what content in https://pytorch.org/docs/stable/index.html is an issue. If this has to do with the general https://pytorch.org website, please file an issue at https://github.com/pytorch/pytorch.github.io/issues/new/choose  instead. If this has to do with https://pytorch.org/tutorials, please file an issue at https://github.com/pytorch/tutorials/issues/new.
```

File: .github/workflows/_docs.yml [hunk 7]
```diff
@@ -194,7 +194,7 @@ jobs:
...
+          path: docs/main/
```
Comment (malfet) [inline .github/workflows/_docs.yml:197]:
```suggestion
          path: pytorch.github.io/docs/main/
```

File: .ci/pytorch/python_doc_push_script.sh [hunk 8]
```diff
@@ -26,8 +26,8 @@ echo "error: python_doc_push_script.sh: version (arg2) not specified"
 fi
 
 # Argument 1: Where to copy the built documentation to
-# (pytorch.github.io/$install_path)
-install_path="${1:-${DOCS_INSTALL_PATH:-docs/${DOCS_VERSION}}}"
+# (docs/$install_path)
```
Comment (malfet) [inline .ci/pytorch/python_doc_push_script.sh:29]:
```suggestion
# (pytorch_docs/$install_path)
```

File: .ci/pytorch/cpp_doc_push_script.sh [hunk 9]
```diff
@@ -23,7 +23,7 @@ echo "error: cpp_doc_push_script.sh: version (arg2) not specified"
 fi
 
 # Argument 1: Where to copy the built documentation for Python API to
-# (pytorch.github.io/$install_path)
+# (docs/$install_path)
```
Comment (malfet) [inline .ci/pytorch/cpp_doc_push_script.sh:26]:
```suggestion
# (pytorch.github.io/$install_path)
```

**Review by malfet — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/documentation.yml [hunk 10]
```diff
@@ -6,7 +6,7 @@ body:
   attributes:
     label: 📚 The doc issue
     description: >
-      A clear and concise description of what content in https://pytorch.org/docs/stable/index.html is an issue. If this has to do with the general https://pytorch.org website, please file an issue at https://github.com/pytorch/pytorch.github.io/issues/new/choose instead. If this has to do with https://pytorch.org/tutorials, please file an issue at https://github.com/pytorch/tutorials/issues/new.
+      A clear and concise description of what content in https://pytorch.org/docs/stable/index.html is an issue. If this has to do with the general https://pytorch.org website, please file an issue at https://github.com/pytorch/pytorch.github.io/issues/new/choose  instead. If this has to do with https://pytorch.org/tutorials, please file an issue at https://github.com/pytorch/tutorials/issues/new.
```
Comment (malfet) [inline .github/ISSUE_TEMPLATE/documentation.yml:9]:
```suggestion
      A clear and concise description of what content in https://pytorch.org/docs/stable/index.html is an issue. If this has to do with the general https://pytorch.org website, please file an issue at https://github.com/pytorch/pytorch.github.io/issues/new/choose instead. If this has to do with https://pytorch.org/tutorials, please file an issue at https://github.com/pytorch/tutorials/issues/new.
```

**Pull request closed without merging**